web/satellite/src/views/Register.vue

```diff
@@ -215,7 +221,46 @@ import { validateEmail } from "@/utils/validation"
 				hasError = true;
 			}
 
+			if(!this.$data.isTermsAccepted){
+                this.$data.isTermsAcceptedError = true;
+                hasError = true;
+			}
+
 			if (hasError) return;
+
+            this.$apollo.mutate({
```

Comment:
we shouldn't use apollo client in each view model. Place this logic in separate file to reduce dependency from apollo